app/src/components/submissionPage/SubmissionTextBox.test.tsx

```diff
@@ -0,0 +1,180 @@
+// SubmissionTextBox.test.tsx
+import React from "react";
+import renderer from "react-test-renderer";
+import { render as librender } from "@testing-library/react";
+import { screen } from "@testing-library/react";
+import { render, unmountComponentAtNode } from "react-dom";
+import { act } from "react-dom/test-utils";
+import MentionElements from "../../types/MentionElements";
+import SubmissionTextBox, { Props } from "./SubmissionTextBox";
+import GetExtraLength from "./GetExtraLength";
+import CharactersLeftDisplay, {
+    GetCharacterCountRemaining,
+} from "./CharactersLeftDisplay";
+import OnSubmitMessageDisplay from "./OnSubmitMessageDisplay";
+import Snap from "../../firebase/snap/Snap";
+import ValidateSnap from "./ValidateSnap";
+import SubmissionBoxErrorDisplay from "./SubmissionBoxErrorDisplay";
+
+let container = null;
+beforeEach(() => {
+    // setup a DOM element as a render target
+    container = document.createElement("div");
+    document.body.appendChild(container);
+});
+
+afterEach(() => {
+    // cleanup on exiting
+    unmountComponentAtNode(container);
+    container.remove();
+    container = null;
+});
+
+it("renders without crashing", () => {
+    act(() => {
+        render(
+            <SubmissionTextBox
+                snappables={[{ id: "test-id", display: "test-display" }]}
+                user={"test-user"}
+            />,
+            container
+        );
+    });
+    expect(container.children.length).toBeGreaterThan(0);
+});
+
+describe("CharactersLeftDisplay", () => {
+    test("characters remaining string renders", () => {
+        const elem1: MentionElements = { id: "idlength10", display: "Jim" };
+        const elem2: MentionElements = {
+            id: "anna@softwire.com",
+            display: "Anna",
+        };
+        const msg2: String =
+            "Hi there @[Anna](anna@softwire.com) and @[Jim](idlength10), you have cool voices!";
+        const component = renderer.create(
+            <CharactersLeftDisplay
+                snappedUsers={[elem1, elem2]}
+                message={msg2}
+            />
+        );
+        let tree = component.toJSON();
+        expect(tree).toMatchSnapshot();
+    });
+    test("characters remaining string renders", () => {
```

Comment:
Test names are important. How does this test differ from the previous one?
If it fails it should be clear what is failing!